### ISSUE TITLE
Question about dusbin.

### ISSUE THREAD

**Issue by limchaos** [rpautrat/SuperPoint#43]
Hi,
Thanks for your code, I have a small question, what is the benefit of using dustbin? Did you try to train the network with out dustbin?

**Comment by rpautrat** [comment]
Hi,

The softmax used before the removal of the dustbin acts as a normalization: the sum of the 65 cells should add up to 1. Now if you don't have a dustbin and there is no interest point in a 8x8 patch, the normalization will still force the network to predict a small probability for the 64 cells (say a probability of 1/64 per cell, which is the default threshold to consider a point as interesting). Whereas if you have the dustbin, the network should ideally predict 0 for the 64 first cells and 1 for the dustbin after the softmax and thus once the dustbin is removed there is no detection at all.

I didn't try without the dustbin but trusted the authors to have compared with/without it and decided that it was better with it.

**Comment by limchaos** [comment]
Hi,
I see, thanks for your explanation, I just did a experiment. For most points, dustbin channel will get the biggest logits, which indicate the fact there is no interesting point in most 8x8 patch.

**Comment by rpautrat** [comment]
In practice it doesn't necessarily mean that there will be no interesting point at all. It's more a way to get rid of a surplus of logits when there is only a few interesting point in the patch. So when the dustbin channel gets the biggest logits, it rather means that there will be only a few points detected in the patch.

**Comment by limchaos** [comment]
Good to know this, thanks.

**Comment by limchaos** [comment]
I should make an inference carefully hahaha

**Comment by limchaos** [comment]
Because the detector head extract interesting point from 8*8 patches, so there will be a problem , the same image with different resolutions should have different structure in such a 8*8 patch. That means if I trained the network in low resolution images will hard to generalize to high resolution images.
![or](https://user-images.githubusercontent.com/33613957/53142063-7e987800-35cd-11e9-93b2-5ca9e098a233.png)
input image with size (1440, 1080)
![s](https://user-images.githubusercontent.com/33613957/53141837-a0ddc600-35cc-11e9-87f2-80659d6b9a27.png)
the heat map of input image (1440, 1080 )
![s8](https://user-images.githubusercontent.com/33613957/53141875-cb2f8380-35cc-11e9-8c1c-d508b48b289f.png)
the heat map of input image down sampled by the scale of 8 (180, 135)   
I trained detector head by the image size of (120, 160)

**Comment by rpautrat** [comment]
Yes that is correct! That's why the synthetic shapes are generated at multiple scales, so that the 8x8 patches can cover the interesting points at various scales. Similarly, the homography adaptation step modifies the scale of the training images and the network is thus trained on multiple scales.

However I agree that the amount of scale used during the homographic adaptation is not so big, so if you train on small images and then evaluate on images 8 times bigger the performance might not be as good as on small images.

But one possible solution would be to downsize your big test images to the same size as was used during training, use MagicPoint on this reduced image and then upscale the features to get features for the original big image. The precision would probably not be perfect, but the detections should still be acceptable.
